pkg/yurthub/autonomymanager/autonomy_manager.go

```diff
@@ -0,0 +1,227 @@
+// 2. Dispatch keys to recover routine
+// 3. Recover routine get object for api server
+// 4. Write it to cache manager and refresh error key
+// utilize queue and wait
+func (am *AutonomyManager) Run(ctx context.Context) {
+	// 1.Consistency Validate every 20 seconds
+	go wait.JitterUntil(func() {
+		if am.restConfigMgr == nil {
+			return
+		}
+		config := am.restConfigMgr.GetRestConfig(true)
+		if config == nil {
+			return
+		}
+		client, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			return
+		}
+		am.ConsistencyValidate(client)
+	}, 20*time.Second, 2, true, ctx.Done())
+
+	// 2. Sync
+	go wait.JitterUntil(func() {
+		if am.restConfigMgr == nil {
+			return
+		}
+		config := am.restConfigMgr.GetRestConfig(true)
+		if config == nil {
+			return
+		}
+		dynamicClient, err := dynamic.NewForConfig(config)
+		if err != nil {
+			return
+		}
+		am.Sync(dynamicClient)
+	}, 20*time.Second, 3, true, ctx.Done())
+
+	// 3. Update errorKeys
+	observer := am.cacheMgr.ObserveErrorKeys()
+	go am.syncErrorKeys(ctx, observer)
+}
+
+// ConsistencyValidate check the data consistency between cache manager and api server
+func (am *AutonomyManager) ConsistencyValidate(client kubernetes.Interface) {
+	klog.Info("start to check the data consistency between cache manager and api server")
+	node, err := client.CoreV1().Nodes().Get(context.TODO(), am.nodeName, metav1.GetOptions{})
+	if err != nil {
+		klog.Warningf("failed to get node; %v", err)
+		return
+	}
+	if node.Annotations == nil || node.Annotations[projectinfo.GetAutonomyAnnotation()] == "false" {
+		// reset node autonomy condition
+		err := EnsureAutonomyCondition(client, node, []v1.ConditionStatus{v1.ConditionUnknown, v1.ConditionTrue}, v1.ConditionFalse,
+			"Autonomy is disabled", "The annotation for enabling autonomy has been canceled or"+
+				"users don't enable autonomy")
+		if err != nil {
+			klog.Warningf("failed to reset node autonomy condition; %v", err)
+		}
+		return
+	}
+	if len(am.errorKeys) == 0 {
+		err = EnsureAutonomyCondition(client, node, []v1.ConditionStatus{v1.ConditionUnknown, v1.ConditionFalse}, v1.ConditionTrue,
+			"Data synchronized", "Autonomy manager has recovered the autonomy ability of node.")
+		if err != nil {
+			klog.Warningf("failed to update node condition: %v", err)
+			return
+		}
+		am.count = 0
+	} else {
+		am.count++
+		if am.count > 3 {
+			var messageList []string
+			for _, val := range am.errorKeys {
+				messageList = append(messageList, val.Error())
+			}
+			msg := strings.Join(messageList, "/n")
+			err := EnsureAutonomyCondition(client, node, []v1.ConditionStatus{v1.ConditionTrue}, v1.ConditionUnknown,
+				"Cache failed", msg)
+			if err != nil {
+				klog.Warningf("failed to update node condition: %v", err)
+			}
+		}
+	}
+}
+
+func EnsureAutonomyCondition(client kubernetes.Interface, node *v1.Node, oldConditionStatus []v1.ConditionStatus, expectedConditionStatus v1.ConditionStatus, reason, message string) error {
```

Comment:
It is not a good idea to update Node status by Yurthub, how about intercept the node status update request from kubelet and modify the node status condition of this request?